extra/migrate_settings.rb

```diff
@@ -1,108 +1,199 @@
-#!/usr/bin/ruby
-#
-# Script to migrate forema-proxy settings.yml
-# to plugin form
-#
-# Greg Sutcliffe <gsutclif@redhat.com> 2014
+require 'pathname'
+require 'fileutils'
+require 'optparse'
 
-require 'yaml'
+module ::Proxy
+  class Migrations
+    attr_reader :migrations, :past_migrations, :migration_state_file_path
 
-def modules
-  [ :settings, :tftp, :dns, :dhcp, :puppet, :puppetca, :bmc, :chef, :realm ]
-end
+    def initialize(migration_state_file_path, past_migrations = [])
+      @past_migrations = past_migrations
+      @migration_state_file_path = migration_state_file_path
+    end
 
-def known_options
-  {
-    :daemon             => :settings,
-    :daemon_pid         => :settings,
-    :log_file           => :settings,
-    :log_level          => :settings,
-    :port               => :settings,
-    :ssl_ca_file        => :settings,
-    :ssl_certificate    => :settings,
-    :ssl_private_key    => :settings,
-    :trusted_hosts      => :settings,
-    :virsh_network      => :settings,
-    :foreman_url        => :settings,
-    :settings_directory => :settings,
-    :http_port          => :settings,
-    :https_port         => :settings,
-    :use_cache          => :settings,
-    :cache_location     => :settings,
-    :puppetca_use_sudo  => :puppetca,
-    :puppetdir          => :puppetca,
-    :ssldir             => :puppetca,
-    :sudo_command       => :puppetca,
-    :customrun_args     => :puppet,
-    :customrun_cmd      => :puppet,
-    :freeipa_remove_dns => :realm
-  }
-end
+    def load_past_migrations!
+      raise "Migration state file #{@migration_state_file_path} can't be found" unless File.exist?(@migration_state_file_path)
+      @past_migrations = File.read(@migration_state_file_path).split('\n')
+      self
+    end
```

Comment:
No point in continuing if an exception occurred, hence no cleaning up. Re: nice error -- I don't really want to wrap or catch/re-raise exceptions (as the whole thing should be killed on exception anyway). I could add logging, but not sure if it's going to be that useful either...